pkgs/applications/networking/instant-messengers/dino/default.nix

```diff
@@ -73,6 +73,26 @@ stdenv.mkDerivation rec {
     libxkbcommon
   ];
 
+  cmakeFlags = ["-DBUILD_TESTS=yes"];
+
+  doCheck = true;
+  checkPhase = ''
+  echo "================= Running built tests ================="
+  set -x
+  ./xmpp-vala-test
+  TEST1=$?
+  ./signal-protocol-vala-test
+  TEST2=$?
+  set +x
+  if [ $TEST1 != 0 ] || [ $TEST2 != 0 ]; then
+    echo "tests failed"
+    exit 1;
+  else
+    echo "tests succeeded"
+  fi
+  echo "================= /Running built tests ================="
+  '';
```

Comment:
Please indent the multiline string, and add the missing `runHook preCheck` and `runHook postCheck` to the overridden phase.